src/org/opensolaris/opengrok/configuration/Group.java

```diff
@@ -134,6 +134,42 @@ public void addGroup(Group g) {
         return parents;
     }
 
+    /**
+     * Collect all related groups to this group. A related group is
+     * <ul>
+     * <li>any anchestor</li>
+     * <li>any subgroup</li>
+     * <ul>
+     *
+     * @return all collected related groups to this group
+     */
+    public Set<Group> getRelatedGroups() {
+        Set<Group> groupsTmp = new TreeSet<>();
+        groupsTmp.addAll(getDescendants());
+        groupsTmp.addAll(getParents());
+        return groupsTmp;
+    }
+
+    /**
+     * Collect all group's projects and repositories included in this group and
+     * in any subgroup.
+     *
+     * @return all collected projects and repositories
+     */
+    public Set<Project> getAllProjects() {
+        Set<Project> projectsTmp = new TreeSet<>();
+        for (Project p : getRepositories()) {
```

Comment:
the `getRepositories()` name is confusing to me as it clashes with the `Repository` object.